modules/bindversion/bindversion.go

```diff
@@ -0,0 +1,79 @@
+/*
+ * ZDNS Copyright 2016 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package bindversion
+
+import (
+	"github.com/miekg/dns"
+	"github.com/zmap/zdns"
+	"github.com/zmap/zdns/modules/miekg"
+)
+
+// result to be returned by scan of host
+type Result struct {
+	BindVersion string `json:"version,omitempty" groups:"short,normal,long,trace"`
+}
+
+// Per Connection Lookup ======================================================
+//
+type Lookup struct {
+	Factory *RoutineLookupFactory
+	miekg.Lookup
+}
+
+func (s *Lookup) DoLookup(_ string) (interface{}, []interface{}, zdns.Status, error) {
+	var res Result
+	innerRes, trace, status, err := s.DoTxtLookup("VERSION.BIND")
+	if status != zdns.STATUS_NOERROR {
+		return res, trace, status, err
```

Comment:
Do you want to return `res` or `nil` here? I'm not very familiar with the API---are you sure you want an empty result back in the error case?